include/bbcode.php

```diff
@@ -68,49 +68,46 @@ function bb_attachment($Text, $simplehtml = false, $tryoembed = true)
 		$data["image"] = "";
 	}
 
+	$return = '';
 	if ($simplehtml == 7) {
-		$text = style_url_for_mastodon($data["url"]);
+		$return = style_url_for_mastodon($data["url"]);
 	} elseif (($simplehtml != 4) && ($simplehtml != 0)) {
-		$text = sprintf('<a href="%s" target="_blank">%s</a><br>', $data["url"], $data["title"]);
+		$return = sprintf('<a href="%s" target="_blank">%s</a><br>', $data["url"], $data["title"]);
 	} else {
-		if ($simplehtml != 4) {
-			$text = sprintf('<span class="type-%s">', $data["type"]);
-		}
-
-		$oembed = sprintf('[bookmark=%s]%s[/bookmark]', $data['url'], $data['title']);
-		if ($tryoembed) {
-			try {
-				$oembed = OEmbed::getHTML($data['url'], $data['title']);
-			} catch (Exception $e) {
-				// $oembed isn't modified
+		try {
+			if ($tryoembed) {
+				$return = OEmbed::getHTML($data['url'], $data['title']);
+			} else {
+				throw new Exception('OEmbed is disabled for this attachment.');
+			}
+		} catch (Exception $e) {
+			if ($simplehtml != 4) {
```

Comment:
Why has this moved into the "catch"? It seems that this will generate some unbalanced HTML, since in line 105 the ```</span>``` will be added anyway - but the matching ```<span...>``` only inside the catch.

Reply:
Basicaly, we try OEmbed first if we're alowed to, and if it fails or `$tryoembed` is false, then we move on to the HTML fallback. The `catch() {...}` actually goes on until line 107, which mean the produced HTML is balanced whether the OEmbed try is successfull or not.

Reply:
Ah, okay.